src/lib/y2network/interface.rb

```diff
@@ -25,13 +25,39 @@ class Interface
     attr_accessor :description
     # @return [Symbol] Interface type
     attr_accessor :type
+    attr_reader :configured
+    attr_reader :hardware
+
+    # Shortcuts for accessing interfaces' ifcfg options
+    ["STARTMODE", "BOOTPROTO"].each do |ifcfg_option|
+      method_name = ifcfg_option.downcase
+
+      define_method method_name do
+        # when switching to new backend we need as much guards as possible
+        if !configured || config.nil? || config.empty?
+          raise "Trying to read configuration of an unconfigured interface"
```

Comment:
please specify more details in exception at least with "name" of interface.

Reply:
done